features/storage/kvstore/global_api/kvstore_global_api.h

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2018 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef _KVSTORE_STATIC_API
+#define _KVSTORE_STATIC_API
+
+#include "stddef.h"
+#include "stdint.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct _opaque_kv_key_iterator *kv_iterator_t;
+
+#define KV_WRITE_ONCE_FLAG                      (1 << 0)
+#define KV_REQUIRE_CONFIDENTIALITY_FLAG         (1 << 1)
+#define KV_REQUIRE_INTEGRITY_FLAG               (1 << 2)
+#define KV_REQUIRE_REPLAY_PROTECTION_FLAG       (1 << 3)
+
+#define KV_MAX_KEY_LENGTH 128
```

Comment:
Should we add this to the config system?

Reply:
Each configuration must be tested and we actually do not have the means for this kind of tests in Mbed OS, unless we associate a value to a target

Reply:
Reasonable for now. But this contradicts the push this release in other teams to integrate with the config system.